include/kernels/block_transpose.cuh

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include <kernel_helper.h>
+#include <color_spinor_field_order.h>
+#include <color_spinor_field.h>
+#include <shared_memory_cache_helper.h>
+
+namespace quda {
+
+  /**
+      Kernel argument struct
+  */
+  template <class v_t_, class b_t_, bool is_device_, typename vFloat, typename vAccessor, typename bFloat, typename bAccessor, int nSpin_, int nColor_, int nVec_>
+  struct BlockTransposeArg : kernel_param<> {
+    using real = typename mapper<vFloat>::type;
+    static constexpr bool is_device = is_device_;
+    static constexpr int nSpin = nSpin_;
+    static constexpr int nColor = nColor_;
+    static constexpr int nVec = nVec_;
+
+    using v_t = v_t_;
+    using b_t = b_t_;
+
+    vAccessor V;
+    bAccessor B[nVec];
+    int_fastdiv block_y;
+    int volume_cb;
+    int actual_nvec;
+
+    BlockTransposeArg(v_t &V, cvector_ref<b_t> &B_, int block_x, int block_y) :
+      // We need full threadblock
+      kernel_param(dim3((V.VolumeCB() + block_x - 1) / block_x * block_x, (nVec + block_y - 1) / block_y * block_y, V.SiteSubset() * nColor)),
+      V(V), block_y(block_y), volume_cb(V.VolumeCB()), actual_nvec(B_.size())
+    {
+      for (auto i = 0u; i < B_.size(); i++) {
+        B[i] = bAccessor(B_[i]);
+      }
+    }
+  };
+
+  template <typename Arg> struct BlockTransposeKernel {
+    const Arg &arg;
+    constexpr BlockTransposeKernel(const Arg &arg) : arg(arg) {}
+    static constexpr const char *filename() { return KERNEL_FILE; }
+
+    __device__ __host__ inline void operator()(int x_cb, int)
```

Comment:
I'd appreciate a one-line comment on what's being transposed---I'm pretty sure it's making the _outer_ near-null vector index the fastest index and vice-versa, but I've had to re-convince myself enough times that a comment in the code would help!

Reply:
The latest of the branch already has the one-line comment you request.